src/analyze/flavors/lwc/refine-feature.ts

```diff
@@ -0,0 +1,71 @@
+import { AnalyzerVisitContext } from "../../analyzer-visit-context";
+import { ComponentFeatureBase } from "../../types/features/component-feature";
+import { ComponentMember } from "../../types/features/component-member";
+import { Node, ClassDeclaration } from "typescript";
+
+import { ComponentMethod } from "../../types/features/component-method";
+import { AnalyzerFlavor } from "../analyzer-flavor";
+import { isLwcComponent } from "./utils";
+
+// In LWC, the public properties & methods must be tagged with @api
+// everything else becomes protected and not accessible externally
+function hasApiDecorator(node: Node | undefined, context: AnalyzerVisitContext) {
+	const { ts } = context;
+	if (node && node.decorators) {
+		for (const decorator of node.decorators) {
+			const expression = decorator.expression;
+
+			// We find the first decorator calling specific identifier name (found in LWC_PROPERTY_DECORATOR_KINDS)
+			if (ts.isIdentifier(expression)) {
+				const identifier = expression;
+				const kind = identifier.text;
+				if (kind === "api") {
+					return true;
+				}
+			}
+		}
+	}
+	return false;
+}
+
+function findClassDeclaration(node: Node | undefined, { ts }: AnalyzerVisitContext): ClassDeclaration | undefined {
+	while (node) {
+		if (ts.isClassDeclaration(node)) {
+			return node;
+		}
+		node = node.parent;
+	}
+}
+
+function isLWCComponent(component: ComponentFeatureBase, context: AnalyzerVisitContext) {
+	const node = findClassDeclaration(component.declaration?.node, context);
+	if (node) {
+		return !!isLwcComponent(node, context);
+	}
+	// How do we know that we are dealing with LWC components?
+	// Currently assume that it is always the case
+	return true;
+}
```

Comment:
Note that I'll be investigating better ergonomics for making flavors with `WCA` . I'll remember to tag you when I need input for how to improve this area :+1:

Reply:
👍